code/photometry.py

```diff
@@ -0,0 +1,171 @@
+import sys
+import os
+import warnings
+from astropy.io import fits
+from astropy.stats import sigma_clipped_stats
+from astropy.wcs import WCS
+import numpy as np
+
+# temporarily let the notebook start without tractor as dependency
+try:
+    from tractor import (Tractor, PointSource, PixPos, Flux, PixelizedPSF, NullWCS,
+                         NullPhotoCal, ConstantSky, Image)
+
+    from find_nconfsources import find_nconfsources
+except ImportError:
+    print("tractor is missing")
+    pass
+
+from exceptions import CutoutError
+from extract_cutout import extract_cutout
+
+
+def setup_for_tractor(band_configs, ra, dec, stype, ks_flux_aper2, infiles, df):
+    # tractor doesn't need the entire image, just a small region around the object of interest
+    subimage, x1, y1, subimage_wcs, bgsubimage = make_cutouts(
+        ra, dec, infiles, band_configs.cutout_width, band_configs.mosaic_pix_scale
+    )
+
+    # set up the source list by finding neighboring sources
+    objsrc, nconfsrcs = find_nconfsources(
+        ra, dec, stype, ks_flux_aper2, x1, y1, band_configs.cutout_width, subimage_wcs, df
+    )
+
+    # measure sky noise and mean level
+    # suppress warnings about nans in the calculation
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore", category=RuntimeWarning)
+        skymean, skymedian, skynoise = sigma_clipped_stats(
+            bgsubimage, sigma=3.0)
+
+    return subimage, objsrc, nconfsrcs, skymean, skynoise
+
+
+def make_cutouts(ra, dec, infiles, cutout_width, mosaic_pix_scale):
```

Comment:
I feel it would be beneficial to enforce the arguments for all these functions in `code` to be keyword only, or at least the ones after ra/dec; overall that would make the notebook code more explanatory and less error-prone.

Reply:
I fixed this in all the files touched by this PR, and added the rest as a to-do in #51.